client/benchmark/src/main/java/org/elasticsearch/client/benchmark/rest/RestClientBenchmark.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.benchmark.rest;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpStatus;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.client.StandardHttpRequestRetryHandler;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.benchmark.AbstractBenchmark;
+import org.elasticsearch.client.benchmark.ops.bulk.BulkRequestExecutor;
+import org.elasticsearch.client.benchmark.ops.search.SearchRequestExecutor;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+public final class RestClientBenchmark extends AbstractBenchmark<RestClient> {
+    public static void main(String[] args) throws Exception {
+        RestClientBenchmark b = new RestClientBenchmark();
+        b.run(args);
+    }
+
+    @Override
+    protected RestClient client(String benchmarkTargetHost) {
+        CloseableHttpClient httpClient = HttpClients
+            .custom()
+            .setRetryHandler(new StandardHttpRequestRetryHandler(3, true))
+            .build();
+
+        return RestClient.builder(new HttpHost(benchmarkTargetHost, 9200))
+            .setHttpClient(httpClient)
```

Comment:
but when we create the http client in RestClient we provide our own connection manager. Anyways,  it doesn't matter cause with https://github.com/elastic/elasticsearch/pull/19400 we are going to have a different http client impl. But we should benchmark our default settings rather than having a different setup for the benchmarks :)